website/scripts/build-static-content.js

```diff
@@ -457,6 +455,12 @@ module.exports = {
                 sails.helpers.strings.random.with({len:10})// if two files in different folders happen to have the same filename, there is a 1/16^10 chance of a collision (this is small enough- worst case, the build fails at the uniqueness check and we rerun it.)
               ).replace(/[^a-z0-9\-]/ig,'');
 
+              // Assert uniqueness of URL paths.
+              if (rootRelativeUrlPathsSeen.includes(rootRelativeUrlPath)) {
+                throw new Error('Failed compiling markdown content: Files as currently named would result in colliding (duplicate) URLs for the website.  To resolve, rename the pages whose names are too similar.  Duplicate detected: ' + rootRelativeUrlPath);
+              }//•
+              rootRelativeUrlPathsSeen.push(rootRelativeUrlPath);
+
```

Comment:
I'd move this up slightly to put it before where we generate the htmlId, so that that the code about deciding the htmlId can stay right next to the code that determines the HTML output path and writes it.